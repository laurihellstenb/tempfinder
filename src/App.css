.App {
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.alert {
  width: 300px;
  padding-bottom: 0;
  margin-top: 1em;
  margin-left: auto;
  margin-right: auto;
}

.searchBar {
  margin-top: 1em;
}

.searchResult {
  margin-top: 1em;
}

.List table {
  border-collapse: collapse;
  width: 600px; 
  margin: auto;
}

.List th, td {
  padding: 8px;
  text-align: left;
  border-bottom: 1px solid #ddd;
}

.List tr:hover {
  background-color:#f5f5f5;
}

.removeButton:hover {
  cursor: pointer;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
