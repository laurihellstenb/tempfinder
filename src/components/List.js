import React from 'react';

const List = ({ cities, delFromList }) => {

    return (
        cities && cities.length > 0 ?
            <div className="List">
                <table>
                    <tbody>
                        <tr><th>City</th><th>Temp</th><th></th></tr>
                        {cities.map((item, index) =>
                            <tr key={index}>
                                <td>{item.name}</td>
                                <td>{item.temp}&#8451;</td>
                                <td><div className="removeButton" id={index} onClick={delFromList}>&#10007;</div>
                                </td>
                            </tr>
                        )}
                    </tbody>
                </table>
            </div> : null
    )
}

export default List;