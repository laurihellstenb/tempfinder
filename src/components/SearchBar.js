import React from 'react';

const SearchBar = ({ handleChange, fetchData, searchterm }) => {

    return (
        <div className="searchBar">
            <input type="text" placeholder="Search for a city..." onChange={handleChange} />
            <button onClick={fetchData} value={searchterm}>Search</button>
        </div>
    )
}

export default SearchBar;

