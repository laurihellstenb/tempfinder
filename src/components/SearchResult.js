import React from 'react';

const SearchResult = ({ name, temp, addToList }) => {

    return (
        name && temp ?
            <div className="searchResult">
                <p>The temperature in {name} is currently {temp}&#8451;  <button onClick={addToList}>Save</button></p>
            </div> : null
    )
}

export default SearchResult;

