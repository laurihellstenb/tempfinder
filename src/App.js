import React, { Component } from 'react';
import './App.css';
import List from './components/List';
import SearchBar from './components/SearchBar';
import SearchResult from './components/SearchResult';
import Alert from 'react-bootstrap/Alert';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = { searchterm: '', cities: [], temp: 0, name: '', show: false }
  }

  fetchData = () => {
    this.setState({ show: false })
    const url = 'https://api.openweathermap.org/data/2.5/weather?q=' + this.state.searchterm + '&units=metric&APIkey=c031dccbb8f243d787c1d4235679da96';
    fetch(url)
      .then(res => res.json())
      .then(resData => {
        this.setState({ name: resData.name, temp: resData.main.temp })
      })
      .catch(error => {
        console.error(error)
        this.setState({ show: true })
      })
  }

  addtoList = () => {
    const newCityObj = {
      name: this.state.name,
      temp: this.state.temp
    }
    if (newCityObj.name !== '') {
      this.setState({ cities: [...this.state.cities, newCityObj] })
    }
  }

  delFromList = (e) => {
    const index = parseInt(e.target.id)
    this.setState({
      cities: this.state.cities.filter((city, i) => i !== index)
    })
  }

  handleChange = (e) => {
    this.setState({ searchterm: e.target.value })
  }

  render() {
    return (
      <div className="App">
        <h2>Temperature finder</h2>
        <SearchBar searchterm={this.state.searchterm} fetchData={this.fetchData} handleChange={this.handleChange} />
        {this.state.show &&
          <Alert className="alert" dismissible variant="warning">
            <p>Could not find location.</p>
          </Alert>}
        <SearchResult name={this.state.name} temp={this.state.temp} addToList={this.addtoList} />
        <List cities={this.state.cities} delFromList={this.delFromList} />
      </div>
    );
  }
}

export default App;
