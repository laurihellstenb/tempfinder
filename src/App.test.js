import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';
import List from './components/List';
import SearchBar from './components/SearchBar';
import SearchResult from './components/SearchResult';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<List />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<SearchBar />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<SearchResult />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  const tree = renderer
    .create(<List />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  const tree = renderer
    .create(<SearchBar />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  const tree = renderer
    .create(<SearchResult />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
