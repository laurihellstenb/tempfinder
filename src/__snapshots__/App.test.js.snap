// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <h2>
    Temperature finder
  </h2>
  <div
    className="searchBar"
  >
    <input
      onChange={[Function]}
      placeholder="Search for a city..."
      type="text"
    />
    <button
      onClick={[Function]}
      value=""
    >
      Search
    </button>
  </div>
</div>
`;

exports[`renders correctly 2`] = `null`;

exports[`renders correctly 3`] = `
<div
  className="searchBar"
>
  <input
    placeholder="Search for a city..."
    type="text"
  />
  <button>
    Search
  </button>
</div>
`;

exports[`renders correctly 4`] = `null`;
